src/voice/udp/handle.rs

```diff
@@ -0,0 +1,256 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::sync::Arc;
+
+use crypto_secretbox::{
+    aead::Aead, cipher::generic_array::GenericArray, KeyInit, XSalsa20Poly1305,
+};
+use discortp::Packet;
+
+use getrandom::getrandom;
+use log::*;
+
+use tokio::{sync::Mutex, sync::RwLock};
+
+use super::UdpSocket;
+
+use crate::{
+    errors::VoiceUdpError,
+    types::VoiceEncryptionMode,
+    voice::{crypto::get_xsalsa20_poly1305_nonce, voice_data::VoiceData},
+};
+
+use super::{events::VoiceUDPEvents, RTP_HEADER_SIZE};
+
+/// Handle to a voice udp connection
```

Comment:
<!--
 Copyright (c) 2024 star
 
 This Source Code Form is subject to the terms of the Mozilla Public
 License, v. 2.0. If a copy of the MPL was not distributed with this
 file, You can obtain one at https://mozilla.org/MPL/2.0/.
-->


```suggestion
/// Handle to a voice UDP connection
```